### PR TITLE
State managment

### PR DESCRIPTION
**Pull request by dawidzon**
I would like to ask you to check my homework.

### CODE REVIEW

File: src/services/post.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-console */
```
Comment (wozny1989) [inline src/services/post.js:1]: ✂️ 
```suggestion
```

**Review by wozny1989 — APPROVED** [review]
@dawidzon LGTM 👍

File: src/pages/PostDetails/PostDetails.jsx [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
-      .then((data) => {
-        setPost(data)
-      })
-      .finally(() => setLoading(false))
-  }, [])
+  const {
+    isLoading,
+    data: post,
+    error,
+  } = useQuery('post', () => postService.fetchPost(postId))
```
Comment (moraw9) [inline src/pages/PostDetails/PostDetails.jsx]: Query id has to be unique, so follow [this pattern](https://react-query-v3.tanstack.com/guides/query-keys#if-your-query-function-depends-on-a-variable-include-it-in-your-query-key).

File: src/pages/PostDetails/PostDetails.jsx [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
+
+  if (error) {
+    return <h1>{error.message}</h1>
```
Comment (moraw9) [inline src/pages/PostDetails/PostDetails.jsx]: Please use also  navigation to `NotFound` page, when not found error is returned

**Review by moraw9 — CHANGES_REQUESTED** [review]
Good, only a few comments

File: src/pages/PostDetails/PostDetails.jsx [hunk 4]
```diff
@@ -1,26 +1,29 @@
...
-      .finally(() => setLoading(false))
-  }, [])
+  const {
+    isLoading,
+    data: post,
+    error,
+  } = useQuery(['post', postId], () => postService.fetchPost(postId))
+
+  const navigate = useNavigate()
+  const handleBackHome = () => navigate('/not-found')
```
Comment (moraw9) [inline src/pages/PostDetails/PostDetails.jsx]: Function name suggests that you are navigated to hame, not to not found page, please change name

Comment (moraw9) [reply src/pages/PostDetails/PostDetails.jsx]: And also not each error is related with not existing resource, so you need to handle navigation to not found page only when error is related with it

File: src/pages/PostDetails/PostDetails.jsx [hunk 5]
```diff
@@ -1,26 +1,29 @@
...
+  }
 
   return (
     <>
-      {loading && <Loading />}
-      {!loading && (
+      {isLoading && <Loading />}
+      {!isLoading && (
         <>
           <h1>{post.title}</h1>
```
Comment (moraw9) [inline src/pages/PostDetails/PostDetails.jsx]: Don't you need to use `?.` after `post` ? What with initial rendering? I'm not sure if object exists.

File: src/pages/PostsList/PostsList.jsx [hunk 6]
```diff
@@ -1,27 +1,29 @@
...
-    setLoading(true)
-    fetch('http://localhost:3000/posts')
-      .then((response) => response.json())
-      .then((data) => {
-        setPosts(data)
-      })
-      .finally(() => setLoading(false))
-  }, [])
+  const navigate = useNavigate()
+  const handleBackHome = () => navigate('/not-found')
```
Comment (moraw9) [inline src/pages/PostsList/PostsList.jsx]: The same as above with `PostDetails`

**Review by moraw9 — CHANGES_REQUESTED** [review]

**Comment by dawidzon** [comment]
I'm not sure if these solutions are correct, but it is working

File: src/pages/PostDetails/PostDetails.jsx [hunk 7]
```diff
@@ -1,29 +1,32 @@
...
+  const {
+    isLoading,
+    data: post,
+    error,
+  } = useQuery(['post', postId], () => postService.fetchPost(postId))
+
+  const navigate = useNavigate()
+  const handleNotFoundPage = () => navigate('/not-found')
+
+  if (error?.message === 'Not found') {
```
Comment (moraw9) [inline src/pages/PostDetails/PostDetails.jsx]: Good! But also please display something if error is different than `Not found`

File: src/pages/PostsList/PostsList.jsx [hunk 8]
```diff
@@ -1,27 +1,29 @@
...
-      .then((response) => response.json())
-      .then((data) => {
-        setPosts(data)
-      })
-      .finally(() => setLoading(false))
-  }, [])
+  const navigate = useNavigate()
+  const handleNotFoundPage = () => navigate('/not-found')
+
+  if (error?.message === 'Not found') {
```
Comment (moraw9) [inline src/pages/PostsList/PostsList.jsx]: The same as above :D

**Review by moraw9 — CHANGES_REQUESTED** [review]
Great, only one small change to do! 👍

File: src/pages/PostDetails/PostDetails.jsx [hunk 9]
```diff
@@ -16,8 +16,12 @@ const PostDetails = () => {
   const navigate = useNavigate()
   const handleNotFoundPage = () => navigate('/not-found')
 
-  if (error?.message === 'Not found') {
-    handleNotFoundPage()
+  if (error) {
+    if (error?.message === 'Not found') {
+      handleNotFoundPage()
+    } else if (error?.message !== 'Not found') {
```
Comment (moraw9) [inline src/pages/PostDetails/PostDetails.jsx]: Simply `else` is enough.

File: src/pages/PostsList/PostsList.jsx [hunk 10]
```diff
@@ -16,8 +16,12 @@ const PostsList = () => {
   const navigate = useNavigate()
   const handleNotFoundPage = () => navigate('/not-found')
 
-  if (error?.message === 'Not found') {
-    handleNotFoundPage()
+  if (error) {
+    if (error?.message === 'Not found') {
+      handleNotFoundPage()
+    } else if (error?.message !== 'Not found') {
```
Comment (moraw9) [inline src/pages/PostsList/PostsList.jsx]: Like above

**Review by moraw9 — CHANGES_REQUESTED** [review]
Only one small change to do :D

**Review by moraw9 — APPROVED** [review]

**Pull request merged**